### PR TITLE
Add TauHybridCSolver to get_best_solver

### PR DESCRIPTION
**Pull request by seanebum**
added TauHybridCSolver into get_best_solver model method and gave it priority over the numpy version

### CODE REVIEW

File: gillespy2/core/model.py [hunk 1]
```diff
@@ -857,28 +857,34 @@ def get_best_solver(self):
...
+
+        if can_use_cpp and hybrid_check and chybrid_check:
+            from gillespy2 import TauHybridCSolver
+            return TauHybridCSolver
+        elif can_use_numpy and hybrid_check:
             from gillespy2 import TauHybridSolver
             return TauHybridSolver
-
         elif not can_use_numpy and hybrid_check:
             raise ModelError('TauHybridSolver is the only solver currently that supports '
```
Comment (briandrawert) [inline gillespy2/core/model.py:884]: "TauHybridSolver is the only solver" should this message be updated?

File: gillespy2/core/model.py [hunk 2]
```diff
@@ -857,28 +857,34 @@ def get_best_solver(self):
...
                              'AssignmentRules, RateRules, FunctionDefinitions, or Events. '
                              'Please install Numpy.')
```
Comment (briandrawert) [inline gillespy2/core/model.py:886]: 'Please install Numpy.'  This is probably not the right solution.  Needs to be revised

Comment (briandrawert) [inline gillespy2/core/model.py:886]: 'Please install Numpy.'  This is probably not the right solution.  Needs to be revised

**Comment by briandrawert** [comment]
Please add "Hybrid" to `get_best_solver_algo()`

File: gillespy2/core/model.py (hunk 2 above)
Comment (seanebum) [reply gillespy2/core/model.py]: Updated this to a pre-check that raises an error if not can_use_cpp and not can_use_numpy that just notifies the user of a general dependency error.  It should, however, never reach this point.

**Review by BryanRumsey — APPROVED** [review]

**Review by BryanRumsey — APPROVED** [review]

**Pull request merged**